scripts/docgen.lua

```diff
@@ -85,7 +85,11 @@ local function require_all_configs()
   -- Configs are lazy-loaded, tickle them to populate the `configs` singleton.
   for _, v in ipairs(vim.fn.glob('lua/lspconfig/server_configurations/*.lua', 1, 1)) do
     local module_name = v:gsub('.*/', ''):gsub('%.lua$', '')
-    configs[module_name] = require('lspconfig.server_configurations.' .. module_name)
+    -- sumneko_lua has been renamed to lua_language_server
+    -- this is a deprecated alias
+    if module_name ~= 'sumneko_lua' then
```

Comment:
does this needed ? doc can be use `lua-language-server` directly.

Reply:
without this, the script creates sections for sumneo_lua (in addition to the lua-language-server section) in server_configurations, which could be misleading to users

Reply:
the doc of `sumneko_lua` also no need ? a special condition looks wired. but it should do in next pr or after release

Reply:
I agree this is a little ugly, but without this both `lua_language_server` and `sumneko_lua` are shown in `doc/server_configurations.{md,txt}` with no suggestion that the latter has been deprecated, and the diff would look like a new language server named `lua_language_server` is added.

Maybe I can do something to this instead? that will keep the aliases in one place:
https://github.com/neovim/nvim-lspconfig/blob/d3c82d2f9a6fd91ec1ffee645664d2cc57e706d9/lua/lspconfig.lua#L13

> the doc of sumneko_lua also no need?

I'm not entirely sure what that means